VENTouchLock/VENTouchLock.m

```diff
@@ -174,71 +180,57 @@ - (void)requestTouchIDWithCompletion:(void (^)(VENTouchLockTouchIDResponse))comp
 
 - (void)lock
 {
-    if (![self isPasscodeSet]) {
+    if (![self isPasscodeSet] || self.lockWindow.isKeyWindow) {
         return;
     }
 
     if (self.splashViewControllerClass != NULL) {
         VENTouchLockSplashViewController *splashViewController = [[self.splashViewControllerClass alloc] init];
         if ([splashViewController isKindOfClass:[VENTouchLockSplashViewController class]]) {
-            UIWindow *mainWindow = [[UIApplication sharedApplication].windows firstObject];
-            UIViewController *rootViewController = [UIViewController ventouchlock_topMostController];
-            UIViewController *displayController;
+            self.lockWindow = [[UIWindow alloc] initWithFrame:self.mainWindow.frame];
```

Comment:
I think this will give us the benefit of overlaying other windows which reside at lower, more normal levels.